components/hubspot/sources/new-contact-property-change/new-contact-property-change.mjs

```diff
@@ -38,27 +39,84 @@ export default {
       const ts = this.getTs(contact);
       return {
         id: `${id}${ts}`,
-        summary: `${properties.firstname} ${properties.lastname}`,
+        summary: properties[this.property],
         ts,
       };
     },
     isRelevant(contact, updatedAfter) {
       return !updatedAfter || this.getTs(contact) > updatedAfter;
     },
-    getParams() {
+    getParams(after) {
       return {
+        object: "contacts",
         limit: 50,
+        properties: [
+          this.property,
+        ],
         sorts: [
           {
             propertyName: "lastmodifieddate",
             direction: "DESCENDING",
           },
         ],
-        propertiesWithHistory: this.property,
+        filterGroups: [
+          {
+            filters: [
+              {
+                propertyName: this.property,
+                operator: "HAS_PROPERTY",
+              },
+              {
+                propertyName: "lastmodifieddate",
+                operator: "GTE",
+                value: after,
+              },
+            ],
+          },
+        ],
       };
     },
+    async batchGetContacts(inputs) {
+      return this.hubspot.makeRequest(
+        API_PATH.CRMV3,
+        "/objects/contacts/batch/read",
+        {
+          method: "POST",
+          data: {
+            properties: [
+              this.property,
+            ],
+            propertiesWithHistory: [
+              this.property,
+            ],
+            inputs,
+          },
+        },
+      );
+    },
     async processResults(after, params) {
-      const { results } = await this.hubspot.listObjectsInPage("contacts", null, params);
+      const updatedContacts = [];
+      do {
+        const {
+          results, paging,
+        } = await this.hubspot.searchCRM(params);
+        updatedContacts.push(...results);
+        if (paging) {
+          params.after = paging.next.after;
+        } else {
+          delete params.after;
+        }
+      } while (params.after);
```

Comment:
This whole pagination block could be a shared method, since it's used a few times on different components in the same fashion.

It can be a common method imported by these components, and could take the method name and params to pass `(this.hubspot.searchCRM, params)`, and return the array of items.

So something like:
```
async paginate(request, params) {
  const arrItems = [];
  do {
    const {
      results, paging,
    } = await request(params);
    arrItems.push(...results);
    params.after = paging?.next.after;
  } while (params.after);
  return arrItems;
}
```
```suggestion
      const updatedContacts = await this.paginate(this.hubspot.searchCRM, params);
```